### PR TITLE
Fix failure with read-only /dev in spec

### PR DESCRIPTION
**Pull request by kolyshkin**
Commit fb4c27c4b79c7d (PR #2570, went into v1.0.0-rc93) fixed a bug with
read-only tmpfs, but introduced a bug with read-only /dev (#3248).

This happens because /dev is a tmpfs mount and is therefore remounted
read-only a bit earlier than before.

To fix,

1. Revert the part of the above commit which remounts all tmpfs mounts
   as read-only in mountToRootfs.

2. Reuse finalizeRootfs (which is already used to remount /dev
   read-only) to also remount all ro tmpfs mounts that were previously
   mounted rw in mountPropagate.

Add a test case to validate the fix and prevent future regressions;
make sure it fails before the fix:
```
 ✗ runc run [ro /dev mount]
   (in test file tests/integration/mounts.bats, line 45)
     `[ "$status" -eq 0 ]' failed
   runc spec (status=0):

   runc run test_busybox (status=1):
   time="2021-11-12T12:19:48-08:00" level=error msg="runc run failed: unable to start container process: error during container init: error mounting \"devpts\" to rootfs at \"/dev/pts\": mkdir /tmp/bats-run-VJXQk7/runc.0Fj70w/bundle/rootfs/dev/pts: read-only file system"
```
Fixes: #3248

1.0 backport: #3277

### CODE REVIEW

**Review by thaJeztah — APPROVED** [review]
LGTM (left one minor suggestion)

File: libcontainer/rootfs_linux.go [hunk 1]
```diff
@@ -164,15 +164,14 @@ func prepareRootfs(pipe io.ReadWriter, iConfig *initConfig, mountFds []int) (err
 // finalizeRootfs sets anything to ro if necessary. You must call
 // prepareRootfs first.
 func finalizeRootfs(config *configs.Config) (err error) {
-	// remount dev as ro if specified
+	// Remount /dev and tmpfs mounts as ro if specified.
 	for _, m := range config.Mounts {
-		if utils.CleanPath(m.Destination) == "/dev" {
-			if m.Flags&unix.MS_RDONLY == unix.MS_RDONLY {
+		if m.Flags&unix.MS_RDONLY == unix.MS_RDONLY {
```
[the remark this answers is not in the document]
Comment (kolyshkin) [reply libcontainer/rootfs_linux.go]: I have also noticed these ifs can be merged into one, and have it a try. Did the same version as yours number two and did not like it as it's too bulky.

I like your first version though, thanks @thaJeztah! Implemented.

**Review by thaJeztah — APPROVED** [review]
LGTM!

**Comment by kolyshkin** [comment]
@AkihiroSuda PTAL

**Review by cyphar — APPROVED** [review]
LGTM

**Pull request closed without merging**